Java Operator crashing on startup after upgrading to Quarkus 1.12.1

Version of the Operator SDK : 1.7.2

Platform : 
- Debian Buster
- openjdk version "11.0.9" 2020-10-20
- OpenJDK Runtime Environment GraalVM CE 20.3.0 (build 11.0.9+10-jvmci-20.3-b06 

The steps taken to cause the bug :

1.  Upgrade from Quarkus 1.10.1 to 1.12.1 
2.  mvn quarkus:dev

`2021-03-05 10:54:05,544 ERROR [io.qua.run.boo.StartupActionImpl] (Quarkus Main Thread) Error running Quarkus: java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at io.quarkus.runner.bootstrap.StartupActionImpl$3.run(StartupActionImpl.java:134)
        at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.NoSuchMethodError: 'io.fabric8.kubernetes.client.dsl.MixedOperation io.fabric8.kubernetes.client.V1ApiextensionAPIGroupDSL.customResourceDefinitions()'
        at io.javaoperatorsdk.operator.Operator.register(Operator.java:106)
        at io.javaoperatorsdk.operator.Operator.register(Operator.java:63)
        at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
        at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
        at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:658)
        at io.javaoperatorsdk.quarkus.extension.OperatorProducer.operator(OperatorProducer.java:24)
        at io.javaoperatorsdk.quarkus.extension.OperatorProducer_ProducerMethod_operator_a8440c868181d0f10deca15fb7a236e1362600c2_Bean.create(OperatorProducer_ProducerMethod_operator_a8440c868181d0f10deca15fb7a236e1362600c2_Bean.zig:334)
        at io.javaoperatorsdk.quarkus.extension.OperatorProducer_ProducerMethod_operator_a8440c868181d0f10deca15fb7a236e1362600c2_Bean.create(OperatorProducer_ProducerMethod_operator_a8440c868181d0f10deca15fb7a236e1362600c2_Bean.zig:349)
        at io.quarkus.arc.impl.AbstractSharedContext.createInstanceHandle(AbstractSharedContext.java:96)
        at io.quarkus.arc.impl.AbstractSharedContext.access$000(AbstractSharedContext.java:14)
        at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:29)
        at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:26)
        at io.quarkus.arc.impl.LazyValue.get(LazyValue.java:26)
        at io.quarkus.arc.impl.ComputingCache.computeIfAbsent(ComputingCache.java:69)
        at io.quarkus.arc.impl.AbstractSharedContext.get(AbstractSharedContext.java:26)
        at io.javaoperatorsdk.quarkus.extension.OperatorProducer_ProducerMethod_operator_a8440c868181d0f10deca15fb7a236e1362600c2_Bean.get(OperatorProducer_ProducerMethod_operator_a8440c868181d0f10deca15fb7a236e1362600c2_Bean.zig:381)
        at io.javaoperatorsdk.quarkus.extension.OperatorProducer_ProducerMethod_operator_a8440c868181d0f10deca15fb7a236e1362600c2_Bean.get(OperatorProducer_ProducerMethod_operator_a8440c868181d0f10deca15fb7a236e1362600c2_Bean.zig:397)
        at com.fruit.QuarkusOperator_Bean.create(QuarkusOperator_Bean.zig:304)
        at com.fruit.QuarkusOperator_Bean.create(QuarkusOperator_Bean.zig:344)
        at io.quarkus.arc.impl.AbstractSharedContext.createInstanceHandle(AbstractSharedContext.java:96)
        at io.quarkus.arc.impl.AbstractSharedContext.access$000(AbstractSharedContext.java:14)
        at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:29)
        at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:26)
        at io.quarkus.arc.impl.LazyValue.get(LazyValue.java:26)
        at io.quarkus.arc.impl.ComputingCache.computeIfAbsent(ComputingCache.java:69)
        at io.quarkus.arc.impl.AbstractSharedContext.get(AbstractSharedContext.java:26)
        at io.quarkus.arc.impl.ClientProxies.getApplicationScopedDelegate(ClientProxies.java:17)
        at com.fruit.QuarkusOperator_ClientProxy.arc$delegate(QuarkusOperator_ClientProxy.zig:67)
        at com.fruit.QuarkusOperator_ClientProxy.run(QuarkusOperator_ClientProxy.zig:126)
        at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:122)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:66)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:42)
        at com.fruit.QuarkusOperator.main(QuarkusOperator.java:26)`

Quarkus libs : 1.12.1.Final
Kubernetes Client and Model libs : 5.1.1
Java Operator libs : 1.7.2

Thank you in advance for any hint.

adding

```
<dependencyManagement>
    <dependencies>
      <dependency>
        <groupId>io.fabric8</groupId>
        <artifactId>kubernetes-client-bom</artifactId>
        <version>5.1.0</version>
        <type>pom</type>
        <scope>import</scope>
    </dependency>
```
works but it downgrades kubernetes-client regarding quarkus platform latest version.

Using quarkus 1.12.0 make use of 5.1.0 but then we encounter another issue

```
java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.javaoperatorsdk.quarkus.extension.deployment.QuarkusExtensionProcessor#createConfigurationServiceAndOperator threw an exception: java.lang.IllegalArgumentException: Cannot instantiate 'org.mhildenb.javaoperator.tutorial.CustomService' CR class
        at io.javaoperatorsdk.quarkus.extension.deployment.QuarkusExtensionProcessor.createControllerConfiguration(QuarkusExtensionProcessor.java:129)
        at io.javaoperatorsdk.quarkus.extension.deployment.QuarkusExtensionProcessor.lambda$createConfigurationServiceAndOperator$1(QuarkusExtensionProcessor.java:80)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
        at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
        at java.base/java.util.HashMap$KeySpliterator.forEachRemaining(HashMap.java:1603)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
        at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
        at io.javaoperatorsdk.quarkus.extension.deployment.QuarkusExtensionProcessor.createConfigurationServiceAndOperator(QuarkusExtensionProcessor.java:81)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:920)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2415)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
        at java.base/java.lang.Thread.run(Thread.java:834)
        at org.jboss.threads.JBossThread.run(JBossThread.java:501)
Caused by: java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
        at io.javaoperatorsdk.quarkus.extension.deployment.QuarkusExtensionProcessor.createControllerConfiguration(QuarkusExtensionProcessor.java:127)
        ... 20 more
Caused by: java.lang.IllegalArgumentException: Cannot instantiate Spec, consider overriding initSpec: org.mhildenb.javaoperator.tutorial.ServiceSpec
        at io.fabric8.kubernetes.client.CustomResource.genericInit(CustomResource.java:342)
        at io.fabric8.kubernetes.client.CustomResource.initSpec(CustomResource.java:117)
        at io.fabric8.kubernetes.client.CustomResource.<init>(CustomResource.java:108)
        at org.mhildenb.javaoperator.tutorial.CustomService.<init>(CustomService.java:10)
        ... 25 more
```

Would it be possible to release a 1.7.3 to align operator sdk on quarkus platform 1.12.1 and fabric.io 5.1.1 ?

--- a/operator-framework-quarkus-extension/pom.xml
+++ b/operator-framework-quarkus-extension/pom.xml
@@ -19,7 +19,7 @@
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
     <maven.compiler.parameters>true</maven.compiler.parameters>
-    <quarkus.version>1.12.0.Final</quarkus.version>
+    <quarkus.version>1.12.1.Final</quarkus.version>
     <compiler-plugin.version>3.8.1</compiler-plugin.version>
     <maven.surefire.version>3.0.0-M5</maven.surefire.version>
   </properties>

--- a/samples/quarkus/pom.xml
+++ b/samples/quarkus/pom.xml
@@ -17,7 +17,7 @@
   <properties>
     <maven.compiler.source>11</maven.compiler.source>
     <maven.compiler.target>11</maven.compiler.target>
-    <quarkus.version>1.12.0.Final</quarkus.version>
+    <quarkus.version>1.12.1.Final</quarkus.version>
     <compiler-plugin.version>3.8.1</compiler-plugin.version>
     <maven.compiler.parameters>true</maven.compiler.parameters>
   </properties>
@@ -78,4 +78,29 @@
       </plugin>
     </plugins>
   </build>
+  
+  <profiles>
+    <profile>
+      <id>native</id>
+      <properties>
+        <quarkus.package.type>native</quarkus.package.type>
+      </properties>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-maven-plugin</artifactId>
+            <version>${quarkus.version}</version>
+            <executions>
+              <execution>
+                <goals>
+                  <goal>build</goal>
+                </goals>
+              </execution>
+            </executions>
+          </plugin>
+        </plugins>
+      </build>
+    </profile>
+  </profiles>
 </project>

